core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3970,11 +3970,18 @@ private void handleReady(long fileId) throws AlluxioException, IOException {
       try (CloseableResource<UnderFileSystem> ufsResource = resolution.acquireUfsResource()) {
         // If previous persist job failed, clean up the temporary file.
         cleanup(ufsResource.get(), tempUfsPath);
+        // if the persist destination is on object store, let persist job copy files to destination
+        // directly
+        if (ServerConfiguration.getBoolean(PropertyKey.UNDERFS_OBJECT_STORE_DIRECT_PERSIST_ENABLED)
```

Comment:
I think the conditional should encompass the cleanup() logic in the previous line. If this property assumes direct writes, we shouldn't be checking to for cleanup of tmp files.

Reply:
this should be good. the cleanup is to remove the left over from previous runs before scheduling a new persist job